File: Backend/src/main/java/com/example/demo/Models/ChatModel.java
package com.example.demo.Models;

import java.util.List;

public class ChatModel {

    private Long id;
    private boolean isHidden1;
    private boolean isHidden2;
    private GroupModel groupModel;
    private List<MessageModel> messages;
    private UserModel user1;
    private UserModel user2;

    public ChatModel() {
    }

    public ChatModel(Long id, boolean isHidden1,boolean isHidden2, GroupModel groupModel, List<MessageModel> messages, UserModel user1, UserModel user2) {
        this.id = id;
        this.isHidden1 = isHidden1;
        this.isHidden2 = isHidden2;
        this.groupModel = groupModel;
        this.messages = messages;
        this.user1 = user1;
        this.user2 = user2;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public boolean isHidden1() {
        return isHidden1;
    }

    public void setHidden1(boolean hidden1) {
        isHidden1 = hidden1;
    }

    public boolean isHidden2() {
        return isHidden2;
    }

    public void setHidden2(boolean hidden2) {
        isHidden2 = hidden2;
    }

    public GroupModel getGroupModel() {
        return groupModel;
    }

    public void setGroupModel(GroupModel groupModel) {
        this.groupModel = groupModel;
    }

    public List<MessageModel> getMessages() {
        return messages;
    }

    public void setMessages(List<MessageModel> messages) {
        this.messages = messages;
    }

    public UserModel getUser1() {
        return user1;
    }

    public void setUser1(UserModel user1) {
        this.user1 = user1;
    }

    public UserModel getUser2() {
        return user2;
    }

    public void setUser2(UserModel user2) {
        this.user2 = user2;
    }
}

File: Frontend/lib/Model/ChatModel.dart
import 'GroupModel.dart';
import 'MessageModel.dart';
import 'UserModel.dart';

class ChatModel {

  String? id;
  bool? isHidden1;
  bool? isHidden2;
  GroupModel? groupModel;
  List<MessageModel>? messages;
  UserModel? user1;
  UserModel? user2;
  bool? unread = true;

  ChatModel(
      {this.id,
      this.isHidden1,
      this.isHidden2,
      this.groupModel,
      this.messages,
      this.user1,
      this.user2,
      this.unread});

  ChatModel.fromJson(Map<String, dynamic> json) {
    id = json["id"].toString();
    isHidden1 = json["isHidden1"];
    isHidden2 = json["isHidden2"];
    if (json["groupModel"] != null) {
      groupModel = GroupModel.fromJson(json["groupModel"]);
    }
    if (json['messages'] != null) {
      messages = <MessageModel>[];
      json['messages'].forEach((v) {
        messages!.add(MessageModel.fromJson(v));
      });
    }
    if (json["user1"] != null) user1 = UserModel.fromJson(json["user1"]);
    if (json["user2"] != null) user2 = UserModel.fromJson(json["user2"]);
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data["id"] = id;
    data["isHidden1"] = isHidden1;
    data["isHidden2"] = isHidden2;
    if (groupModel != null) data['groupModel'] = groupModel!.toJson();
    if (user1 != null) data['user1'] = user1!.toJson();
    if (user2 != null) data['user2'] = user2!.toJson();
    if (messages != null) {
      data['messages'] = messages!.map((v) => v.toJson()).toList();
    }
    return data;
  }

  get getId => id;

  set setId(id) => this.id = id;

  get getIsHidden1 => isHidden1;

  set setIsHidden1(isHidden1) => this.isHidden1 = isHidden1;

  get getIsHidden2 => isHidden2;

  set setIsHidden2(isHidden2) => this.isHidden2 = isHidden2;

  get getGroupModel => groupModel;

  set setGroupModel(groupModel) => this.groupModel = groupModel;

  get getMessages => messages;

  set setMessages(messages) => this.messages = messages;

  get getUser1 => user1;

  set setUser1(user1) => this.user1 = user1;

  get getUser2 => user2;

  set setUser2(user2) => this.user2 = user2;

  get getUnread => unread;

  set setUnread(unread) => this.unread = unread;
}
